File: Ej 1/src/isp1/CarreraMontana.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package isp1;

/**
 *
 * @author joy11
 */
public class CarreraMontana extends Carrera{
    String modalidad = "montaña";
    
    @Override
    public String getModalidad(){
        return modalidad;
    }
    
    public int getnumBicis(){
        return this.bicis.size();
    }
    
    @Override
    public void quitarBicis(){
        int numBicis = this.getnumBicis();
        double quitar = (numBicis * 20)/100;    //El 20 % de las bicicletas de montaña se quitaran en la carrera
        //System.out.println("Numero de bicis a quitar: " + quitar);
        for(int i=0; i < quitar; i++){
            this.bicis.remove(i);
        }
    }
}

File: Ej 2/carrera_montana.py
from carrera import Carrera
import math
from clonable import Clonable

class CarreraMontana(Carrera, Clonable):
    def __init__(self):
        super().__init__()  # Llama al constructor de la clase base para inicializar la lista de bicis
        self.modalidad = "montaña"
    
    def getModalidad(self):
        return self.modalidad
    
    def getnumBicis(self):
        return len(self.bicis)
    
    def quitarBicis(self):
        numBicis = self.getnumBicis()
        quitar = math.floor(numBicis * 0.20)  # Calcula el 20% de las bicicletas, redondeado hacia abajo
        # Elimina el 20% de las bicicletas de la lista
        for _ in range(quitar):
            self.bicis.pop(0)  # asumiendo que se desea quitar las bicis del inicio de la lista